tests/functional/upgrade/test_resources.py

```
@@ -126,13 +126,22 @@ def test_pod_log_after_upgrade():
         f"Check that the log '{expected_log_after_upgrade}' "
         f"appears after the osd/mon/mg pod is initialized"
     )
     for pod_name in pod_names:
         pod_logs = get_pod_logs(pod_name=pod_name, all_containers=True)
-        assert expected_log_after_upgrade in pod_logs, (
+        if expected_log_after_upgrade in pod_logs:
+            continue
+        if early_startup_marker not in pod_logs:
+            log.warning(
+                f"Pod {pod_name}: early startup logs were rotated out by kubelet, "
+                f"cannot verify '{expected_log_after_upgrade}'"
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)